src/hotspot/share/opto/bytecodeInfo.cpp

```diff
@@ -558,7 +570,7 @@ void InlineTree::print_inlining(ciMethod* callee_method, int caller_bci,
 bool InlineTree::ok_to_inline(ciMethod* callee_method, JVMState* jvms, ciCallProfile& profile,
                               bool& should_delay) {
   assert(callee_method != NULL, "caller checks for optimized virtual!");
-  assert(!should_delay, "should be initialized to false");
+  assert(!should_delay || AlwaysIncrementalInline, "should be initialized to false");
```

Comment:
I'm not sure how useful this assert is now.  It could be changed to should_delay == AlwaysIncrementalInline, or maybe just removed?

Reply:
Agreed, removed.